dimod/discrete/cppdqm.pxd

```diff
@@ -0,0 +1,97 @@
+        AdjVectorDQM(np.uint32_t*, size_type, np.float64_t*, size_type, np.uint32_t*, np.uint32_t*, np.float64_t*, size_type)
+        AdjVectorDQM(np.uint64_t*, size_type, np.uint32_t*,  size_type, np.uint64_t*, np.uint64_t*, np.uint32_t*,  size_type)
+        AdjVectorDQM(np.uint64_t*, size_type, np.uint64_t*,  size_type, np.uint64_t*, np.uint64_t*, np.uint64_t*,  size_type)
+        AdjVectorDQM(np.uint64_t*, size_type, np.int32_t*,   size_type, np.uint64_t*, np.uint64_t*, np.int32_t*,   size_type)
+        AdjVectorDQM(np.uint64_t*, size_type, np.int64_t*,   size_type, np.uint64_t*, np.uint64_t*, np.int64_t*,   size_type)
+        AdjVectorDQM(np.uint64_t*, size_type, np.float32_t*, size_type, np.uint64_t*, np.uint64_t*, np.float32_t*, size_type)
+        AdjVectorDQM(np.uint64_t*, size_type, np.float64_t*, size_type, np.uint64_t*, np.uint64_t*, np.float64_t*, size_type)
+        AdjVectorDQM(np.int32_t*,  size_type, np.uint32_t*,  size_type, np.int32_t*,  np.int32_t*,  np.uint32_t*,  size_type)
+        AdjVectorDQM(np.int32_t*,  size_type, np.uint64_t*,  size_type, np.int32_t*,  np.int32_t*,  np.uint64_t*,  size_type)
+        AdjVectorDQM(np.int32_t*,  size_type, np.int32_t*,   size_type, np.int32_t*,  np.int32_t*,  np.int32_t*,   size_type)
+        AdjVectorDQM(np.int32_t*,  size_type, np.int64_t*,   size_type, np.int32_t*,  np.int32_t*,  np.int64_t*,   size_type)
+        AdjVectorDQM(np.int32_t*,  size_type, np.float32_t*, size_type, np.int32_t*,  np.int32_t*,  np.float32_t*, size_type)
+        AdjVectorDQM(np.int32_t*,  size_type, np.float64_t*, size_type, np.int32_t*,  np.int32_t*,  np.float64_t*, size_type)
+        AdjVectorDQM(np.int64_t*,  size_type, np.uint32_t*,  size_type, np.int64_t*,  np.int64_t*,  np.uint32_t*,  size_type)
+        AdjVectorDQM(np.int64_t*,  size_type, np.uint64_t*,  size_type, np.int64_t*,  np.int64_t*,  np.uint64_t*,  size_type)
+        AdjVectorDQM(np.int64_t*,  size_type, np.int32_t*,   size_type, np.int64_t*,  np.int64_t*,  np.int32_t*,   size_type)
+        AdjVectorDQM(np.int64_t*,  size_type, np.int64_t*,   size_type, np.int64_t*,  np.int64_t*,  np.int64_t*,   size_type)
+        AdjVectorDQM(np.int64_t*,  size_type, np.float32_t*, size_type, np.int64_t*,  np.int64_t*,  np.float32_t*, size_type)
+        AdjVectorDQM(np.int64_t*,  size_type, np.float64_t*, size_type, np.int64_t*,  np.int64_t*,  np.float64_t*, size_type)
+
+
+        # methods
+
+        bool is_self_loop_present() except +
+        size_type num_variables() except +
+	size_type num_variable_interactions() except +
+	size_type num_cases() except +
+	size_type num_case_interactions() except +
+        bias_type get_linear_case(variable_type, variable_type) except +
+        void set_linear_case(variable_type, variable_type, bias_type) except +
+        void get_linear[io_bias_type](variable_type, io_bias_type*) except +
```

Comment:
io_bias_type / io_variable_type. are used for passing templated pointers, io standing for input output here.